src/ParallelAlgorithms/Amr/Protocols/AmrMetavariables.hpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+#include "ParallelAlgorithms/Amr/Protocols/Mutator.hpp"
+
+namespace amr::protocols {
+/// \brief Compile-time information for AMR mutators
+///
+/// A class conforming to this protocol is placed in the metavariables to
+/// provide the list of AMR mutators (each of which must conform to
+/// amr::protocols::Mutator) that will be applied by:
+/// - amr::Actions::InitializeChild and amr::Actions::InitializeParent in order
+///   to initialize data on newly created elements.
+/// - amr::Actions::AdjustDomain in order to update data on existing elements in
+///   case their Mesh or neighbors have changed.
+///
+///
+/// Here is an example for a class conforming to this protocol:
+///
+/// \snippet Amr/Test_Protocols.cpp amr_mutators
+struct AmrMetavariables {
+  template <typename ConformingType>
+  struct test {
+    using mutators = typename ConformingType::mutators;
```

Comment:
Maybe `mutators` -> `projectors` everywhere? I'm just thinking how we want to call these things. "AMR projectors" seems a bit more descriptive to me than "AMR mutators" (they project data between elements at different resolution, rather than just generically "mutate" things). Then naming the classes `ProjectX` would be reasonable too.

Reply:
sure